app/questionnaire/value_source_resolver.py

```diff
@@ -0,0 +1,124 @@
+from dataclasses import dataclass
+from typing import Optional, Union
+
+from app.data_models import AnswerStore, ListStore
+from app.data_models.list_store import ListModel
+from app.questionnaire import Location, QuestionnaireSchema
+from app.questionnaire.relationship_location import RelationshipLocation
+
+answer_value_types = Union[str, int, float, list, dict, None]
+value_source_types = Union[str, int, float, list, None]
```

Comment:
> - I don't understand why there are two types here. As answer value sources are one of the value source types, surely `value_source_types` should contain everything in `answer_value_types`?

Value sources currently do not support returning a dict object, whereas an answer can. For dict objects, value source expects a selector, i.e line 1 from address dict answer etc.


> - Should the value type be `Decimal` for answer values rather than `float`?

Yes, have changed.

> - We should think about where answer value types is defined as it will be needed in more places. There is already an inline definition used in the `_parse_transforms` method of `placeholder_parser.py`

I have moved answer value types to `answer.py` and updated all the references.